crates/telio-wg/src/wg.rs

```diff
@@ -528,9 +549,22 @@ impl State {
                 let old_state = old.state();
                 let new_state = new.state();
 
+                let old_link_state = self
+                    .additional_peer_info
+                    .get(key)
+                    .map(|p| p.link_state)
+                    .unwrap_or_default();
+                let new_link_state = new_additional_peer_info
+                    .get(key)
+                    .map(|p| p.link_state)
+                    .unwrap_or_default();
+
                 #[allow(clippy::collapsible_if)]
-                if !old.is_same_event(new) || old_state != new_state {
-                    if !try_send(new_state, new.clone()).await {
+                if !old.is_same_event(new)
+                    || old_state != new_state
+                    || old_link_state != new_link_state
+                {
+                    if !try_send(new_state, new_link_state, new.clone()).await {
```

Comment:
I think we should only ever use `link_state` when we consider the peer as `Connected`. This way it should reduce the confusion on API usage part(what does it mean that Link is `up` but State is `down`) for example, as it might happen.

I suggest wrapping `link_state` into an Option and have it as `None` if `State != Connected`.

It should then also only be emitted when the peer is considered as connected, thus not confusing the API user. This should work sort of automatically as a side effect of what's written above :)